engine/runtime/src/test/scala/org/enso/compiler/test/semantic/ImportExportTest.scala

```diff
@@ -940,13 +940,18 @@ class ImportExportTest
         .reason
         .asInstanceOf[errors.ImportExport.AmbiguousImport]
       ambiguousImport.symbolName shouldEqual "A_Type"
-      val baos   = new ByteArrayOutputStream()
-      val stream = new ObjectOutputStream(baos)
       mainIr.preorder.foreach(
         _.passData.prepareForSerialization(langCtx.getCompiler.context)
       )
-      stream.writeObject(mainIr)
-      baos.toByteArray should not be empty
+      try {
+        val arr = Persistance.writeObject(mainIr)
+        fail("Shouldn't return anything when there is an error" + arr)
```

Comment:
I think that we shouldn't be caching erroneous `IR`s. OK, @Akirathan?

Reply:
I am not sure. It depends on where and when is the erroneous IR reported?

Reply:
The erroneous IR is reported while it is being stored. `writeObject(...)` finds an element that is not persistable and yields an `IOException`. E.g. the caches with errors can't even be created.